pathfinding_sdf/pathfinding_sdf.c

```diff
@@ -0,0 +1,670 @@
+    int pathCatEndY = 25;
+    PathfindingNode *mapCat = (PathfindingNode *)MemAlloc(gridWidth * gridHeight * sizeof(PathfindingNode));
+    PathfindingNode *pathCat = (PathfindingNode *)MemAlloc(gridWidth * gridHeight * sizeof(PathfindingNode) * 4);
+    int pathCatLength = 0;
+
+    float walkedPathDistanceRat = 0.0f;
+    float walkedPathDistanceCat = 0.0f;
+
+    // toggle flags for mouse and keyboard input
+    int visualizeMode = 0;
+    int randomizeBlocks = 1;
+    char paintMode = 0;
+    int updateSDF = 1;
+    int sdfFunction = 0;
+    int jumpingEnabled = 1;
+    //--------------------------------------------------------------------------------------
+
+    // Main game loop
+    while (!WindowShouldClose())    // Detect window close button or ESC key
+    {
+        //----------------------------------------------------------------------------------
+        // Draw
+        //----------------------------------------------------------------------------------
+        BeginDrawing();
+
+            ClearBackground((Color) {170,200,150,255});
+
+            Vector2 mousePos = GetMousePosition();
+            int cellX = mousePos.x / cellSize;
+            int cellY = mousePos.y / cellSize;
+
+            //----------------------------------------------------------------------------------
+            // mouse input handling
+            //----------------------------------------------------------------------------------
+            if (IsMouseButtonPressed(MOUSE_LEFT_BUTTON))
+            {
+                paintMode = blockedCells[cellY * gridWidth + cellX] == 1 ? 0 : 1;
+            }
+            if (IsMouseButtonDown(MOUSE_LEFT_BUTTON))
+            {
+                blockedCells[cellY * gridWidth + cellX] = paintMode;
+                updateSDF = 1;
+            }
+
+            //----------------------------------------------------------------------------------
+            // keyboard input handling
+            //----------------------------------------------------------------------------------
+            if (IsKeyDown(KEY_C)) 
+            {
+                for (int i = 0; i < gridWidth * gridHeight; i++)
+                {
+                    blockedCells[i] = 0;
+                }
+                updateSDF = 1;
+            }
+
+            if (IsKeyPressed(KEY_V))
+            {
+                visualizeMode++;
+            }
+
+            if (IsKeyPressed(KEY_Q))
+            {
+                ratWallFactor = (ratWallFactor + 1) % 8;
+                updateSDF = 1;
+            }
+
+            if (IsKeyPressed(KEY_R))
+            {
+                randomizeBlocks = 1;
+            }
+
+            if (IsKeyPressed(KEY_S))
+            {
+                sdfFunction = (sdfFunction + 1) % 3;
+                updateSDF = 1;
+            }
+
+            if (IsKeyPressed(KEY_J))
+            {
+                jumpingEnabled = !jumpingEnabled;
+                updateSDF = 1;
+            }

```

Comment:
Maybe put this into a HandleInput function?

Reply:
Hm. It's true the complexity grew more than I anticipated in the beginning. I'll look into how to do this together with breaking down the loop code.